test/postControllerTest.js

```diff
@@ -0,0 +1,187 @@
+/* global it, describe */
+const express = require('express');
+const request = require('supertest');
+const postController = require('../controllers/postController');
+const sinon = require('sinon');
+const postService = require('../services/postService');
+const assert = require('assert');
+
+const app = express();
+
+app.set('view engine', 'ejs');
+app.use(express.static('../public'));
+
+app.use('', postController);
+
+describe('postController', () => {
+  let stubGetAllPosts = sinon.stub();
+  let stubGetOnePost = sinon.stub();
+  let stubCreateNewPost = sinon.stub();
+
+  beforeEach(() => {
+    stubGetAllPosts = sinon.stub(postService, 'getAllPosts').resolves('expected');
+    stubGetOnePost = sinon.stub(postService, 'getOnePost').resolves('expected');
+    stubCreateNewPost = sinon.stub(postService, 'createNewPost').returns(Promise.resolve('expected'));
+  });
+
+  afterEach(() => {
+    stubGetAllPosts.restore();
+    stubGetOnePost.restore();
+    stubCreateNewPost.restore();
+  });
+
+  describe('GET /', () => {
+    it('should call method "postService.getAllPosts" with no arguments', (done) => {
+      request(app)
+        .get('/')
+        .end((err) => {
+          sinon.assert.calledWith(stubGetAllPosts);
+          if (err) return done(err);
+          return done();
```

Comment:
It's probably more consise to just return the promise rather than repeating `if (err) return done(err); return done();` in every test.

The 4th code block in supertest's [README](https://github.com/visionmedia/supertest) has a good example:
```
describe('GET /users', function() {
  it('respond with json', function() {
    return request(app)
      .get('/users')
      .set('Accept', 'application/json')
      .expect(200)
      .then(response => {
          assert(response.body.email, 'foo@bar.com')
      })
  });
});
```

Reply:
```
.then(() => {
  sinon.assert.calledWith(postService.getAllPosts);
  done();
})
.catch(err => {
  done(err);
});
```
Given that this written correctly, is this more concise and preferrable?

Reply:
That isn't what I'm expecting. I'm expecting that you don't need to call `done` manually... and I think returning the promise (as shown above will work)

Reply:
Are you returning the promise?

Reply:
I'm not sure what you mean. If you mean within the ```.then()``` handler then no. I don't have a promise to return, just a regular assertion like how they have it in the example.

Reply:
In the example I posted above:

Reply:
OH wow, I completely missed that.